File: app/src/main/java/com/glucocontrol/domain/repository/AuthRepository.kt
package com.glucocontrol.domain.repository

import com.glucocontrol.domain.model.User

/**
 * Contrato de autenticación para la capa de dominio.
 * La implementación concreta reside en [AuthRepositoryImpl] (capa de datos).
 */
interface AuthRepository {
    /** Inicia sesión con email y contraseña. Lanza excepción si las credenciales son incorrectas. */
    suspend fun signInWithEmail(email: String, password: String)

    /** Crea una cuenta nueva con email y contraseña. */
    suspend fun signUpWithEmail(email: String, password: String)

    /**
     * Inicia sesión con un ID token de Google obtenido vía Credential Manager.
     * Supabase intercambia este token por una sesión usando el proveedor Google OAuth.
     */
    suspend fun signInWithGoogle(idToken: String)

    /** Cierra la sesión activa y limpia los tokens locales. */
    suspend fun signOut()

    /** Devuelve el usuario autenticado actualmente, o null si no hay sesión. */
    suspend fun getCurrentUser(): User?
}

File: lib/domain/repository/auth_repository.dart
import 'package:glucocontrol/domain/model/user.dart';

abstract class AuthRepository {
  Future<void> signInWithEmail(String email, String password);
  Future<void> signUpWithEmail(String email, String password);
  Future<void> signInWithGoogle();
  Future<void> signOut();
  Future<AppUser?> getCurrentUser();
  Future<void> deleteAccount();
}
